prog/mutation.go

```diff
@@ -11,192 +11,203 @@ import (
 	"github.com/google/syzkaller/sys"
 )
 
-func (p *Prog) Mutate(rs rand.Source, ncalls int, ct *ChoiceTable) {
+func (p *Prog) Mutate(rs rand.Source, ncalls int, ct *ChoiceTable, corpus *[]*Prog) {
```

Comment:
Pass []*Prog instead of *[]*Prog.
Slice is 3 words (pointer, length, capacity), copying slice just copies these words, it does not copy the array itself.
Passing pointer to slice header makes sense only iff the function changes the slice itself.
Note that nil is a valid value for a slice, so you don't need to change call sites that pass nil.